pallets/pallet-bonded-coins/src/curves/lmsr.rs

```diff
@@ -34,13 +34,27 @@ where
 	Parameter: FixedSigned + PartialOrd<Precision> + From<Precision> + ToFixed,
 	<Parameter as Fixed>::Bits: Copy + ToFixed + AddAssign + BitOrAssign + ShlAssign,
 {
-	fn calculate_exp_term(&self, supply: Parameter, x: Parameter) -> Result<Parameter, ArithmeticError> {
-		supply
-			.checked_sub(x)
-			.ok_or(ArithmeticError::Underflow)?
-			.checked_div(self.m)
-			.ok_or(ArithmeticError::DivisionByZero)
-			.and_then(|exponent| exp::<Parameter, Parameter>(exponent).map_err(|_| ArithmeticError::Overflow))
+	fn lse(&self, supply: Vec<Parameter>) -> Result<Parameter, ArithmeticError> {
+		// Find the maximum value in the supply for numerical stability
+		let max = supply.iter().max().ok_or(ArithmeticError::Underflow)?;
```

Comment:
Is `Underflow` because of lack of a better error? Or why?

Reply:
Yeah, it should never happen. There should be a validation check in `create_pool`.

Reply:
Added an error log: [75be462](https://github.com/KILTprotocol/kilt-node/pull/777/commits/75be462022dada2193acf79c528d6118d46e2020)

Reply:
seems like it. if we want to stick to arithmetic errors, the best fit would be DivisionByZero if you ask me, given that an empty vector, just like a vector of 0 issuances, would result in trying to compute a log of 0, which is not defined - just like division by zero.

Reply:
Changed error type in: [462df41](https://github.com/KILTprotocol/kilt-node/pull/777/commits/462df410680064af879bccc1d92586392899492d)